### PR TITLE
feat(stylelint): add no-unknown-token-name

### LINKED ISSUE #64: New Lint Rule: Unknown `--rh` token
#1 Provides an opportunity to add a linting rule for unknown tokens

say we deleted the tokens `--rh-color-black-*` and added `--rh-color-gray-*`

in that case, this stylesheet should fail this rule:

```css
#icon {
  color: var(--rh-color-black-900, #151515);
}
```

we could provide an `migrations` options which would allow us to auto-fix these errors:

```yaml
rhds/unknown-token:
  - true
  - migrations:
    --rh-color-black-400: --rh-color-grey-300
```

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did
Added a new stylelint rule: `rhds/no-unknown-token-name`

Fails:
```css
a {
  padding: var(--rh-space-xx);
}
```

Passes:
```css
a {
  padding: var(--rh-space-xl);
}
```

Closes #64

### CODE REVIEW

File: plugins/stylelint.cjs [hunk 1]
```diff
@@ -1,69 +1,5 @@
...
-};
-
-/** @type {import('stylelint').Plugin} */
-const ruleFunction = (_options, _secondaryOptions, ctx) => {
-  return (root, result) => {
-    const validOptions = stylelint.utils.validateOptions(
-      result,
-      ruleName,
-      {
-        /* .. */
```
Comment (zeroedin) [inline plugins/stylelint.cjs:22]: Any reason to keep these comment lines?  I noticed they are in the plugin documentation but seem like placeholders for optional values we could remove.

File: plugins/stylelint/rules/no-unknown-token-name.cjs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+};
+
+/** @type {import('stylelint').Plugin} */
+const ruleFunction = (_, opts, ctx) => {
+  return (root, result) => {
+    const validOptions = stylelint.utils.validateOptions(
+      result,
+      ruleName,
+      {
+        /* .. */
```
Comment (zeroedin) [inline plugins/stylelint/rules/no-unknown-token-name.cjs:23]: same as previous comment

File: plugins/stylelint/rules/token-values.cjs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+};
+
+/** @type {import('stylelint').Plugin} */
+const ruleFunction = (_options, _secondaryOptions, ctx) => {
+  return (root, result) => {
+    const validOptions = stylelint.utils.validateOptions(
+      result,
+      ruleName,
+      {
+        /* .. */
```
Comment (zeroedin) [inline plugins/stylelint/rules/token-values.cjs:22]: same as previous comment

**Review by zeroedin — APPROVED** [review]
This looks good to me, although I just now learned about stylelint plugins and tape testing vs using mocha.  Tests run, and the code looks to be correct and easy enough to follow given my recent introduction to these tools.

File: plugins/stylelint/rules/no-unknown-token-name.cjs (hunk 2 above)
Comment (bennypowers) [reply plugins/stylelint/rules/no-unknown-token-name.cjs:23]: we can use them later on to add validations. no issue of kb over the wire so might as well keep them

**Pull request merged**